### PR TITLE
[UITest] Implement WaitForIdeIdle to wait for same status message

### PR DESCRIPTION
**Pull request by manish**

### CODE REVIEW

File: main/tests/UserInterfaceTests/Ide.cs [hunk 1]
```diff
@@ -165,6 +165,40 @@ static void PollStatusMessage (string[] statusMessage, int timeoutInSecs, int po
 				}
 			}, pollStep: pollStepInSecs * 1000, timeout: timeoutInSecs * 1000);
 		}
+
+		public static void WaitForIdeIdle (uint idlePeriodInSecs = 10, uint retries = 10)
+		{
```
Comment (kdubau) [inline main/tests/UserInterfaceTests/Ide.cs]: So does this mean it waits `idlePeriodInSecs * retries` seconds for the Ide to become idle?

File: main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs [hunk 2]
```diff
@@ -58,8 +58,8 @@ public void RunASPTest (string templateName, BeforeBuildAction beforeBuild)
 				TemplateKindRoot = GeneralKindRoot,
 				TemplateKind = templateName
 			};
-			CreateBuildProject (templateOptions, beforeBuild.GetAction ());
-			IsTemplateSelected (templateOptions);
+
+			CreateBuildProject (templateOptions, () => Ide.WaitForIdeIdle (retries: 5));
```
Comment (kdubau) [inline main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs]: Wouldn't it be more clear to just provide a timeout in seconds?

File: main/tests/UserInterfaceTests/Ide.cs (hunk 1 above)
Comment (manish) [inline main/tests/UserInterfaceTests/Ide.cs]: Yup. slightly more than that, due to the delegate body.

File: main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs (hunk 2 above)
Comment (manish) [inline main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs]: You want to provide the total timeout and then compute the retries from dividing total timeout by `idlePeriodInSecs`?

Comment (kdubau) [inline main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs]: yea, it would be more obvious to someone reading the code.

Comment (manish) [inline main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs]: okay. Can make the change

File: main/tests/UserInterfaceTests/TemplateTests/DotNetTemplatesTest.cs [hunk 3]
```diff
@@ -35,12 +35,12 @@ public class MonoDevelopTemplatesTest : CreateBuildTemplatesTestBase
...
-		[TestCase ("Gtk# 2.0 Project", BeforeBuildAction.None, TestName = "TestCreateBuildGtkSharp20Project", Description = "Create and build a GTK#2 Project")]
-		[TestCase ("Library", BeforeBuildAction.None, TestName = "TestCreateBuildLibrary", Description = "Create and build a Library Project")]
-		[TestCase ("NUnit Library Project", BeforeBuildAction.WaitForPackageUpdate, TestName = "TestCreateBuildNUnitLibraryProject",
+		[TestCase ("Console Project", 5, TestName = "TestCreateBuildConsoleProject", Description = "Create and build C# Console Project")]
+		[TestCase ("Gtk# 2.0 Project", 5, TestName = "TestCreateBuildGtkSharp20Project", Description = "Create and build a GTK#2 Project")]
+		[TestCase ("Library", 5, TestName = "TestCreateBuildLibrary", Description = "Create and build a Library Project")]
+		[TestCase ("NUnit Library Project", 10, TestName = "TestCreateBuildNUnitLibraryProject",
 			Description = "Create and build NUnit Library Project")]
-		public void RunDotNetTests (string templateName, BeforeBuildAction beforeBuild)
+		public void RunDotNetTests (string templateName, int retriesIdeReady)
```
Comment (kdubau) [inline main/tests/UserInterfaceTests/TemplateTests/DotNetTemplatesTest.cs]: Just pass `waitForIdleTimeout` to the test and call `CreateBuildProject (templateOptions,  () => Ide.WaitForIdeIdle (totalTimeoutInSecs: waitForIdleTimeout));`

Comment (kdubau) [inline main/tests/UserInterfaceTests/TemplateTests/DotNetTemplatesTest.cs]: Keep the "retry" idea internal to `WaitForIdeIdle`

**Comment by kdubau** [comment]
Looks good. Let me test

File: main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs [hunk 4]
```diff
@@ -35,22 +35,22 @@ public class ASPNetTemplatesTest : CreateBuildTemplatesTestBase
...
+		[TestCase ("ASP.NET MVC Project", TestName = "TestASPMVCProject",
 			Description = "Create and build ASP.NET MVC Project")]
-		[TestCase ("ASP.NET MVC Project with Unit Tests", BeforeBuildAction.WaitForPackageUpdate, TestName = "TestASPMVCProjectWithUnitTests",
+		[TestCase ("ASP.NET MVC Project with Unit Tests", TestName = "TestASPMVCProjectWithUnitTests",
 			Description = "Create and build ASP.NET MVC Project with Unit Tests")]
-		[TestCase ("ASP.NET MVC Razor Project", BeforeBuildAction.WaitForPackageUpdate, TestName = "TestASPMVCMazorProject",
+		[TestCase ("ASP.NET MVC Razor Project", TestName = "TestASPMVCMazorProject",
 			Description = "Create and build ASP.NET MVC Razor Project")]
-		[TestCase ("ASP.NET MVC Razor Project with Unit Tests", BeforeBuildAction.WaitForPackageUpdate, TestName = "TestASPMVCMazorProjectWithUnitTests",
-			Description = "Create and build \"ASP.NET MVC Razor Project with Unit Tests")]
```
Comment (kdubau) [inline main/tests/UserInterfaceTests/TemplateTests/ASPNETTemplateTests.cs]: Unrelated but just noticed, why is this random quote here?

**Comment by kdubau** [comment]
:+1:

**Pull request merged**